gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java

```diff
@@ -0,0 +1,270 @@
+  }
+
+  DnsRecord(Builder builder) {
+    this.name = builder.name;
+    this.rrdatas = ImmutableList.copyOf(builder.rrdatas);
+    this.ttl = builder.ttl;
+    this.type = builder.type;
+    this.zoneName = builder.zoneName;
+    this.zoneId = builder.zoneId;
+  }
+
+  /**
+   * Enum for the DNS record types supported by Cloud DNS.
+   *
+   * <p> Google Cloud DNS currently supports records of type A, AAAA, CNAME, MX NAPTR, NS, PTR, SOA,
+   * SPF, SRV, TXT.
+   *
+   * @see <a href="https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types">Cloud DNS
+   * supported record types</a>
+   */
+  public enum DnsRecordType {
+    A,
+    AAAA,
+    CNAME,
+    MX,
+    NAPTR,
+    NS,
+    PTR,
+    SOA,
+    SPF,
+    SRV,
+    TXT;
+  }
+
+  public static class Builder {
+
+    private List<String> rrdatas = new LinkedList<>();
+    private String name;
+    private Integer ttl;
+    private DnsRecordType type;
+    private String zoneName;
+    private Long zoneId;
+
+    private Builder() {
+    }
+
+    /**
+     * Creates a builder and pre-populates attributes with the values from the provided DnsRecord
+     * instance.
+     */
+    public Builder(DnsRecord record) {
+      this.name = record.name;
+      this.ttl = record.ttl;
+      this.type = record.type;
+      this.zoneId = record.zoneId;
+      this.zoneName = record.zoneName;
+      this.rrdatas.addAll(record.rrdatas);
+    }
+
+    /**
+     * Adds a record to the record set. The records should be as defined in RFC 1035 (section 5) and
+     * RFC 1034 (section 3.6.1). Examples of records are available in Google DNS documentation.
+     *
+     * @see <a href="https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types">Google
+     * DNS documentation </a>.
+     */
+    public Builder add(String record) {
+      this.rrdatas.add(checkNotNull(record));
+      return this;
+    }
+
+    /**
+     * Sets name for this DNS record set. For example, www.example.com.
+     */
+    public Builder name(String name) {
+      this.name = checkNotNull(name);
+      return this;
+    }
+
+    /**
+     * Sets the number of seconds that this record can be cached by resolvers. This number must be
+     * non-negative.
+     *
+     * @param ttl A non-negative number of seconds
+     */
+    public Builder ttl(int ttl) {
+      checkArgument(ttl >= 0, "TTL cannot be negative. The supplied value was " + ttl + ".");
+      this.ttl = ttl;
+      return this;
+    }
+
+    /**
+     * The identifier of a supported record type, for example, A, AAAA, MX, TXT, and so on.
+     */
+    public Builder type(DnsRecordType type) {
+      this.type = checkNotNull(type);
+      return this;
+    }
+
+    /**
+     * Builds the DNS record.
+     */
+    public DnsRecord build() {
+      return new DnsRecord(this);
+    }
+
+    /**
+     * Sets references to the managed zone that this DNS record belongs to.
+     */
+    public Builder managedZone(ManagedZoneInfo parent) {
```

Comment:
I see your point. The user would probably never asked to use this. I am making it package private for now.

You are correct that we do not have the ID available when listing. However, if we could get it from somewhere, I would prefer having it set. I added a method <code>Builder managedZone(String zoneName)</code> (package private) and made a todo note for the existing <code>Builder managedZone(ManagedZoneInfo zoneInfo)</code> to consider if we actually need it later when. However, I would like to keep it for now.